### PR TITLE
SISRP-4447 - Write spec to cover PhotoController

### PR DESCRIPTION
**Pull request by winstonkim**
https://jira.berkeley.edu/browse/SISRP-4447

### CODE REVIEW

File: spec/controllers/photo_controller_spec.rb [hunk 1]
```diff
@@ -1,29 +1,34 @@
 require 'spec_helper'
 
 describe PhotoController do
+  let(:user_id) { random_id }
 
   context "when serving a users photo" do
```
Comment (ctweney) [inline spec/controllers/photo_controller_spec.rb]: Please change double quotes to single quotes throughout the file (except where double quotes are needed for string interpolation).

File: spec/controllers/photo_controller_spec.rb [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
-          test_photo_object = {:photo => 'photo_binary_content'}
+          test_photo_object = {'photo' => 'photo_binary_content'}
           CampusOracle::Queries.stub(:get_photo).and_return(test_photo_object)
         end
         it "renders users raw image" do
-          allow(controller).to receive(:send_data).with('photo_binary_content', type: 'image/jpeg', disposition: 'inline').and_return(true)
+          expect(controller).to receive(:send_data).with('photo_binary_content', type: 'image/jpeg', disposition: 'inline'){
+            controller.render nothing: true
+          }
           get :my_photo
```
Comment (ctweney) [inline spec/controllers/photo_controller_spec.rb]: This test needs some expectations!
Something along the lines of 
```
expect(response.status).to eq 200
```
And maybe some checks on the response.body content as well.

File: spec/controllers/photo_controller_spec.rb [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
+        end
       context "when user has photo" do
         before do
-          test_photo_object = {:photo => 'photo_binary_content'}
+          test_photo_object = {'photo' => 'photo_binary_content'}
           CampusOracle::Queries.stub(:get_photo).and_return(test_photo_object)
         end
         it "renders users raw image" do
-          allow(controller).to receive(:send_data).with('photo_binary_content', type: 'image/jpeg', disposition: 'inline').and_return(true)
+          expect(controller).to receive(:send_data).with('photo_binary_content', type: 'image/jpeg', disposition: 'inline'){
```
Comment (ctweney) [inline spec/controllers/photo_controller_spec.rb]: I think this block is not necessary. Set expectations on the response instead.

File: spec/controllers/photo_controller_spec.rb [hunk 4]
```diff
@@ -1,29 +1,34 @@
...
-
-    context "when user is not logged in" do
-      before { session['user_id'] = nil }
-      it "returns 401 error with no body" do
-        get :my_photo
-        expect(response.status).to eq 401
-        expect(response.body).to eq " "
+      context "when user has no photo" do
+        before do
+          test_photo_object = nil
```
Comment (ctweney) [inline spec/controllers/photo_controller_spec.rb]: You can collapse this to a single line: 
```
CampusOracle::Queries.stub(:get_photo).and_return(nil)
```

File: spec/controllers/photo_controller_spec.rb [hunk 5]
```diff
@@ -1,29 +1,34 @@
...
+          CampusOracle::Queries.stub(:get_photo).and_return(test_photo_object)
+        end
+        it "renders users raw image" do
+          allow(controller).to receive(:send_data).with(nil, type: 'image/jpeg', disposition: 'inline').and_return(true)
+          get :my_photo
```
Comment (ctweney) [inline spec/controllers/photo_controller_spec.rb]: Again, the allow() shouldn't be necessary; instead, set expectations about what the response will be.

File: spec/controllers/photo_controller_spec.rb (hunk 4 above)
Comment (paulkerschen) [inline spec/controllers/photo_controller_spec.rb]: @winstonkim I believe that the `.stub` is also deprecated these days; the recommended way to do it now would be `allow(CampusOracle::Queries).to receive(:get_photo).and_return(nil)`.

**Comment by winstonkim** [comment]
@ctweney @paulkerschen Thanks for the comments! I made the changes. Let me know if there is anything else that should be changed.

**Comment by ctweney** [comment]
Travis is half-satisfied. I am completely satisfied.

**Pull request merged**